### PR TITLE
fix formula issue for book

### PR DESCRIPTION
**Pull request by daming-lu**
https://github.com/PaddlePaddle/PaddlePaddle.org/issues/256

### CODE REVIEW

File: portal/deploy/strip.py [hunk 1]
```diff
@@ -110,17 +124,25 @@ def book(generated_documentation_dir, version, output_dir_name):
 
                 # Find the markdown element.
                 markdown_body = soup.select('div#markdown')
+                markdown_body_str = unicode(str(markdown_body[0]), 'utf-8')
+                regex_map = {}
```
Comment (varunarora) [inline portal/deploy/strip.py]: We had planned to change this name. Let's call it `formula_map` or something :)

File: portal/deploy/strip.py [hunk 2]
```diff
@@ -110,17 +124,25 @@ def book(generated_documentation_dir, version, output_dir_name):
...
+                                                extensions=['markdown.extensions.fenced_code',
+                                                            'markdown.extensions.tables']
+                                            ) + '\n{% endverbatim %}'
+
+                        soup = BeautifulSoup(converted_content, 'lxml')
+                        spans = soup.select('.markdown-equation')
+                        for s in spans:
+                            s.string = regex_map[s.get('id')]
+
+                        new_html_partial.write(soup.prettify())
```
Comment (varunarora) [inline portal/deploy/strip.py]: `prettify` causes formatting problems, let's not do that

File: portal/deploy/strip.py [hunk 3]
```diff
@@ -110,17 +124,25 @@ def book(generated_documentation_dir, version, output_dir_name):
 
                 # Find the markdown element.
                 markdown_body = soup.select('div#markdown')
+                markdown_body_str = unicode(str(markdown_body[0]), 'utf-8')
+                regex_map = {}
+                markdown_body_str = reserve_formulas(markdown_body_str, regex_map)
```
Comment (varunarora) [inline portal/deploy/strip.py]: A couple of sentences of comments on what this does and why it is needed would be very useful!

File: portal/deploy/strip.py [hunk 4]
```diff
@@ -110,17 +124,25 @@ def book(generated_documentation_dir, version, output_dir_name):
...
                 if len(markdown_body) > 0:
                     with codecs.open(new_path, 'w', 'utf-8') as new_html_partial:
-                        # Strip out the wrapping HTML
-                        new_html_partial.write(
-                            '{% verbatim %}\n' + markdown.markdown(
-                                '\n'.join(unicode(str(markdown_body[0]), 'utf-8').split('\n')[1:-2]),
-                                extensions=['markdown.extensions.fenced_code', 'markdown.extensions.tables']
-                            ) + '\n{% endverbatim %}'
-                        )
+                        converted_content = '{% verbatim %}\n' + markdown.markdown(
```
Comment (varunarora) [inline portal/deploy/strip.py]: Attach `{% verbatim %}` during the `.write` call

File: portal/deploy/strip.py [hunk 5]
```diff
@@ -110,17 +124,25 @@ def book(generated_documentation_dir, version, output_dir_name):
...
+                                                '\n'.join(markdown_body_str.split('\n')[1:-2]),
+                                                extensions=['markdown.extensions.fenced_code',
+                                                            'markdown.extensions.tables']
+                                            ) + '\n{% endverbatim %}'
+
+                        soup = BeautifulSoup(converted_content, 'lxml')
+                        spans = soup.select('.markdown-equation')
```
Comment (varunarora) [inline portal/deploy/strip.py]: `spans` is a terrible variable name! :laughing:

File: portal/deploy/strip.py (hunk 2 above)
Comment (varunarora) [reply portal/deploy/strip.py]: Checkout the other places where the BeautifulSoup output is just written as a string

File: portal/deploy/strip.py (hunk 1 above)
Comment (daming-lu) [reply portal/deploy/strip.py]: Done

File: portal/deploy/strip.py (hunk 3 above)
Comment (daming-lu) [reply portal/deploy/strip.py]: Done

File: portal/deploy/strip.py (hunk 4 above)
Comment (daming-lu) [reply portal/deploy/strip.py]: Done

File: portal/deploy/strip.py (hunk 5 above)
Comment (daming-lu) [reply portal/deploy/strip.py]: Done

File: portal/deploy/strip.py (hunk 2 above)
Comment (daming-lu) [reply portal/deploy/strip.py]: Done

**Pull request merged**